### PR TITLE
Software Training SKale

### PR DESCRIPTION
**Pull request by sahil-kale**

### CODE REVIEW

File: src/turtle_setter.cpp [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+};
+
+static Waypoint movingTurtleWP; //static to keep it file-scope
+
+float distance(float x, float y)
+{
+    return pow(pow(x,2) + pow(y,2), 1/float(2));
+}
+
+float arctan(float x, float y)
```
Comment (azumnanji) [inline src/turtle_setter.cpp]: c++ already has an arctan func with cmath, why not use that?

File: src/turtle_setter.cpp [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    spawnService.request.y = y;
+    spawnService.request.theta = angle;
+    return ros::service::call("spawn", spawnService);
+}
+
+/******CLEAR & SPAWN TURTLE FUNCTIONS END******/
+
+bool resetTurtlePosition(software_training_assignment::resetTurtle::Request &request, software_training_assignment::resetTurtle::Response &response)
+{
+    const double RESET_X = 25;
```
Comment (azumnanji) [inline src/turtle_setter.cpp]: if these are const declare them globally as `constexpr double`. should be 25.0 and 10.0 as well for clarity

File: src/turtle_setter.cpp [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    software_training_assignment::SeekTurtleFeedback feedback;
+    software_training_assignment::SeekTurtleResult result;
+
+    geometry_msgs::Twist msg;
+    
+    const float timeInterval = 1; //arbitary time interval?
+    ros::Rate rate(1/timeInterval);
+    const float totalTime = 10;
+    //Declared volatile since I'm not sure how exactly goal is updated...?
+    float volatile deltaX = movingTurtleWP.x - goal->x;
```
Comment (azumnanji) [inline src/turtle_setter.cpp]: doesn't need to be volatile. deltaX cannot change outside of this function and is only updated here so no need for it

File: src/turtle_setter.cpp [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+void move_turtle_callBack(const software_training_assignment::SeekTurtleGoalConstPtr &goal, Server *actionServer)
+{
+    software_training_assignment::SeekTurtleFeedback feedback;
+    software_training_assignment::SeekTurtleResult result;
+
+    geometry_msgs::Twist msg;
+    
+    const float timeInterval = 1; //arbitary time interval?
+    ros::Rate rate(1/timeInterval);
+    const float totalTime = 10;
```
Comment (azumnanji) [inline src/turtle_setter.cpp]: avoid const inside of functions. const is unnecessary here

File: src/turtle_setter.cpp [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    //Declared volatile since I'm not sure how exactly goal is updated...?
+    float volatile deltaX = movingTurtleWP.x - goal->x;
+    float volatile deltaY= movingTurtleWP.y - goal->y;
```
Comment (azumnanji) [inline src/turtle_setter.cpp]: same as above

File: src/turtle_setter.cpp (hunk 1 above)
Comment (sahil-kale) [reply src/turtle_setter.cpp]: The function here calculates the angle from 0 using arctan. I use the cmath function, but this function name is misleading. I will update it :)

File: src/turtle_setter.cpp (hunk 2 above)
Comment (sahil-kale) [reply src/turtle_setter.cpp]: Sounds good!

File: src/turtle_setter.cpp [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+    spawnService.request.theta = angle;
+    return ros::service::call("spawn", spawnService);
+}
+
+/******CLEAR & SPAWN TURTLE FUNCTIONS END******/
+
+bool resetTurtlePosition(software_training_assignment::resetTurtle::Request &request, software_training_assignment::resetTurtle::Response &response)
+{
+    
+    std::string name = request.name;  //QUESTION: Why do I not need to dereference this??
```
Comment (azumnanji) [inline src/turtle_setter.cpp:99]: this is not a pointer, its passed by reference and therefore you can access the members directly

Comment (sahil-kale) [reply src/turtle_setter.cpp:99]: Ah I see, thanks!!

**Pull request closed without merging**